src/components/backend-ai-settings-view.ts

```diff
@@ -286,16 +323,16 @@ export default class BackendAiSettingsView extends BackendAIPage {
     });
   }
 
-  toggleImageUpdate(e) {
-    if (e.target.checked === false) {
-      globalThis.backendaiclient.setting.set('docker/image/auto_pull', 'none').then((response) => {
-        console.log(response);
-      });
-    } else {
-      globalThis.backendaiclient.setting.set('docker/image/auto_pull', 'digest').then((response) => {
+  setImagePullingBehavior(e) {
+    if (e.target.selected === null) return false;
+    if (e.target.selected.value !== this.options['image_pulling_behavior'] && ['none', 'digest', 'tag'].includes(e.target.selected.value)) {
+      globalThis.backendaiclient.setting.set('docker/image/auto_pull', e.target.selected.value).then((response) => {
+        this.notification.text = _text("notification.SuccessfullyUpdated");
+        this.notification.show();
```

Comment:
I think `this.options['image_pulling_behavior']` should be updated with `e.target.selected.value` after successfully setting `docker/image/auto_pull`. Otherwise, the update will not work properly if you select a different option and then select the previous value again due to the checks in `if` statement.

Reply:
Fixed.